aspnetcore/grpc/authn-and-authz.md

```diff
@@ -0,0 +1,143 @@
+
+    // ... access data from ClaimsPrincipal ...
+}
+
+```
+
+### Bearer token authentication
+
+The client can provide an access token for authentication. The server validates the token and uses it to identify the user.
+
+On the server, bearer token authentication is configured using the [JWT Bearer middleware](/dotnet/api/microsoft.extensions.dependencyinjection.jwtbearerextensions.addjwtbearer).
+
+In the .NET gRPC client, the token can be sent with calls as a header:
+
+```cs
+public bool DoAuthenticatedCall(
+    Ticketer.TicketerClient client, string token)
+{
+    var headers = new Metadata();
+    headers.Add("Authorization", $"Bearer {token}");
+
+    var request = new BuyTicketsRequest { Count = 1 };
+    var response = await client.BuyTicketsAsync(request, headers);
+
+    return response.Success;
+}
+```
+
+### Client certificate authentication
+
+A client could alternatively provide a client certificate for authentication. [Certificate authentication](https://tools.ietf.org/html/rfc5246#section-7.4.4) happens at the TLS level, long before it ever gets to ASP.NET Core. When the request enters ASP.NET Core [Microsoft.AspNetCore.Authentication.Certificate](xref:security/authentication/certauth) allows you to resolve the certificate to a `ClaimsPrincipal`.
+
+In the .NET gRPC client, the client certificate is added to `HttpClientHandler` that is then used to create the gRPC client:
+
+```cs
+public Ticketer.TicketerClient CreateClientWithCert(
+    string baseAddress,
+    X509Certificate2 certificate)
+{
+    // Add client cert to the handler
+    var handler = new HttpClientHandler();
+    handler.ClientCertificates.Add(certificate);
+
+    // Create the gRPC client
+    var httpClient = new HttpClient(handler);
+    httpClient.BaseAddress = new Uri(baseAddress);
+
+    return GrpcClient.Create<Ticketer.TicketerClient>(httpClient);
+}
+```
+
+## Authorize users to access services and service methods
+
+By default, all methods in a service can be called by unauthenticated users. In order to require authentication, apply the [Authorize](/dotnet/api/microsoft.aspnetcore.authorization.authorizeattribute) attribute to the service:
+
+```cs
+[Authorize]
+public class TicketerService : Ticketer.TicketerBase
+{
+}
+```
+
+You can use the constructor arguments and properties of the `[Authorize]` attribute to restrict access to only users matching specific [authorization policies](xref:security/authorization/policies). For example, if you have a custom authorization policy called `MyAuthorizationPolicy` you can ensure that only users matching that policy can access the service using the following code:
+
+```csharp
+[Authorize("MyAuthorizationPolicy")]
+public class TicketerService : Ticketer.TicketerBase
+{
+}
+```
+
+Individual service methods can have the `[Authorize]` attribute applied as well. If the current user doesn't match the policy applied to the method, an error is returned to the caller:
```

Comment:
```suggestion
Individual service methods can have the `[Authorize]` attribute applied as well. If the current user doesn't match the policies applied to **both** the method and the class, an error is returned to the caller:
```

(That's my understanding of how this would work, if that's not the case, a different clarification would be needed)